integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml

```diff
@@ -0,0 +1,29 @@
+#
+# Copyright (c) 2021 IBM Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nested-configmap-secret-pod
+spec:
+  terminationGracePeriodSeconds: 0
+  runtimeClassName: kata
+  containers:
+    - name: test-container
+      image: quay.io/prometheus/busybox:latest
+      command: ["tail", "-f", "/dev/null"]
+      volumeMounts:
+        - mountPath: /config
+          name: config
+        - mountPath: /config/secret
+          name: secret
+  volumes:
+    - name: secret
+      secret:
+        secretName: secret
+    - name: config
+      configMap:
+        name: config
```

Comment:
Please could you consider removing the hard-coded values here by using `envsubst(1)`. If you renamed `pod-nested-configmap-secret.yaml` to `pod-nested-configmap-secret.yaml.in`, you could then change `pod-nested-configmap-secret.yaml.in` to be similar to:

```diff
diff --git a/integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml b/integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml
index 018fe358..07592e23 100644
--- a/integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml
+++ b/integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml
@@ -17,13 +17,13 @@ spec:
       volumeMounts:
         - mountPath: /config
           name: config
-        - mountPath: /config/secret
+        - mountPath: /${configmap_name}/${secret_name}
           name: secret
   volumes:
     - name: secret
       secret:
-        secretName: secret
+        secretName: ${secret_name}
     - name: config
       configMap:
-        name: config
+        name: ${configmap_name}
   restartPolicy: Never
```

Then, the bats script can generate the final YAML file like this:

```bash
envsubst \
  < integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml.in \
  > integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml
```

/cc @wainersm. We need to put that in the the shell script doc when we write it (https://github.com/kata-containers/kata-containers/issues/3128 :)

Reply:
+1 for this suggestion. I'd like to just add one thing: you don't necessarily need to create a temporary file (the `integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml` file in the example above), as you can send the output directly to `kubectl`. For example,
```
envsubst \
  < integration/kubernetes/runtimeclass_workloads/pod-nested-configmap-secret.yaml.in | kubectl -f -
```

@jodh-intel I added the use of `envsubst` in the list of advises in  https://github.com/kata-containers/kata-containers/issues/3128

Reply:
@wainersm - Nice: simpler is better! ;)

Reply:
If we use `.in` as the file extension we lose some of the editor niceties and validation when it detects the file as yaml. Perhaps using a `.template.yaml` or `.tmpl.yaml` suffix would work better.

In some cases there is value in creating the file so you can do the `create` in the setup and a matching `delete` in the teardown. Let me update to show what I'm thinking.

Reply:
Here's what it looked like -- https://github.com/skaegi/tests/commit/b6aadfef674b7e3478eadd2c45a49c0118555eaa -- it works but it's actually fairly complex. I found you had to `export` all the shell variables and also had to be careful to quote values in your yaml.